### PR TITLE
DOCSP-40791: Insert documents guide

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-40791
Staging - https://preview-mongodbnorareidy.gatsbyjs.io/cpp-driver/DOCSP-40791-insert-guide/write/insert/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/write/insert.txt [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+and assign the following values to your ``db`` and ``collection`` variables:
+
+.. literalinclude:: /includes/write/insert.cpp
+    :language: cpp
+    :dedent:
+    :start-after: start-db-coll
+    :end-before: end-db-coll
+
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see
+the :ref:`<cpp-get-started>` tutorial.
```
Comment (norareidy) [inline source/write/insert.txt:45]: Note to reviewer: this link will be broken until #13 is merged

**Review by rustagir — APPROVED** [review]
lgtm with a few comments

File: source/includes/write/insert.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    // end-insert-many
+
+    // Inserts multiple documents and instructs the insert operation to skip document-level validation
+    // start-modify
+    std::vector<bsoncxx::document::value> docs;
+    docs.push_back(make_document(kvp("name", "Mongo's Burgers")));
+    docs.push_back(make_document(kvp("name", "Mongo's Pizza")));
+    docs.push_back(make_document(kvp("name", "Mongo's Tacos")));
+
+    mongocxx::v_noabi::options::insert opts;
```
Comment (kevinAlbs) [inline source/includes/write/insert.cpp:46]:
```suggestion
    mongocxx::options::insert opts;
```

Suggest leaving `v_noabi` out of code examples. Added a suggestion with rationale to [C++ Driver Code Example Patterns and Style Reference](https://docs.google.com/document/d/1wBsUnT69a7XNTUOZd1agQae_pL1fREl-mcg34QgYaCs/edit).

Suggestion applies to other references to `v_noabi` namespaces in this PR.

**Review by kevinAlbs — CHANGES_REQUESTED** [review]
Suggested to remove `v_noabi`. Requesting additional review from @eramongodb as he may be more familiar with the ABI namespace expectations.

**Review by eramongodb — CHANGES_REQUESTED** [review]
I support the omission of `::v_noabi`. Root namespace declarations should be compatible across ABI versions unless an API major version is released which break API compatibility (which would almost certainly require greater changes to documentation anyways). We should encourage use of root namespace declarations by default; explicit ABI namespaces should only be used by code that has known ABI stability requirements.

File: source/write/insert.txt [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+API Documentation
+~~~~~~~~~~~~~~~~~
+
+To learn more about any of the methods or types discussed in this
+guide, see the following API documentation:
+
+- `insert_one() <{+api+}/classmongocxx_1_1v__noabi_1_1collection.html#af040f1fcb1ac406037bf1cbcdb300061>`__
+- `insert_many() <{+api+}/classmongocxx_1_1v__noabi_1_1collection.html#a61ebaa0c63296761637ce45115512085>`__
+- `mongocxx::v_noabi::options::insert <{+api+}/classmongocxx_1_1v__noabi_1_1options_1_1insert.html>`__
```
Comment (eramongodb) [inline source/write/insert.txt:175]: Aside: I _would_ recommend links to API reference pages also use root namespace declarations, but Doxygen [doesn't appear to support](https://mongocxx.org/api/mongocxx-3.9.0/classmongocxx_1_1options_1_1insert.html) such a link to a "redirect". 😢 More investigation may be required as part of documentation-related CXX-2745 work (alongside the yet-unresolved Doxygen using-declaration issue).

Comment (norareidy) [reply source/write/insert.txt:175]: Yeah, removing the v_noabi causes the link to break unfortunately

**Review by eramongodb — APPROVED** [review]

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**